packages/lodestar/src/chain/chain.ts

```diff
@@ -328,35 +331,51 @@ export class BeaconChain implements IBeaconChain {
    * Seeds the fork choice with an anchor state.
    * This state is set as the finalized state.
    */
-  private async initForkChoice(anchorState: TreeBacked<BeaconState>): Promise<void> {
-    let blockRoot;
-    let justifiedCheckpoint;
-    let finalizedCheckpoint;
+  private async initForkChoice(
+    anchorState: TreeBacked<BeaconState>
+  ): Promise<{forkChoice: ForkChoice; checkpoint: Checkpoint}> {
     if (anchorState.latestBlockHeader.slot === GENESIS_SLOT) {
       const block = getEmptyBlock();
       block.stateRoot = this.config.types.BeaconState.hashTreeRoot(anchorState);
-      blockRoot = this.config.types.BeaconBlock.hashTreeRoot(block);
+      const blockRoot = this.config.types.BeaconBlock.hashTreeRoot(block);
       const blockCheckpoint = {
         root: blockRoot,
         epoch: computeEpochAtSlot(this.config, anchorState.slot),
       };
-      justifiedCheckpoint = blockCheckpoint;
-      finalizedCheckpoint = blockCheckpoint;
+      const store = new ForkChoiceStore({
+        emitter: this.emitter,
+        currentSlot: this.clock.currentSlot,
+        justifiedCheckpoint: blockCheckpoint,
+        finalizedCheckpoint: blockCheckpoint,
+      });
+      return {
+        forkChoice: ForkChoice.fromGenesis(this.config, store, block),
+        checkpoint: blockCheckpoint,
+      };
     } else {
       const blockHeader = this.config.types.BeaconBlockHeader.clone(anchorState.latestBlockHeader);
-      blockHeader.stateRoot = this.config.types.BeaconState.hashTreeRoot(anchorState);
-      blockRoot = this.config.types.BeaconBlockHeader.hashTreeRoot(blockHeader);
-      justifiedCheckpoint = anchorState.currentJustifiedCheckpoint;
-      finalizedCheckpoint = anchorState.finalizedCheckpoint;
+      if (this.config.types.Root.equals(blockHeader.stateRoot, ZERO_HASH)) {
+        blockHeader.stateRoot = anchorState.hashTreeRoot();
+      }
+      const finalizedCheckpoint = {
+        root: this.config.types.BeaconBlockHeader.hashTreeRoot(blockHeader),
+        epoch: computeEpochAtSlot(this.config, anchorState.slot),
+      };
+      const justifiedCheckpoint = {
+        root: this.config.types.BeaconBlockHeader.hashTreeRoot(blockHeader),
+        epoch: computeEpochAtSlot(this.config, anchorState.slot) + 1,
```

Comment:
what's the reason to add 1 to justified checkpoint epoch?

This is most likely not the correct justified checkpoint for a specific state but a reasonable one to start with and forkchoice will update to the correct one later.

I think if we just leave this justified checkpoint `epoch` the same to finalized checkpoint `epoch`, we'll update to correct justified checkpoint sooner due to the condition `if state.current_justified_checkpoint.epoch > store.justified_checkpoint.epoch:` in https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/fork-choice.md#on_block

Reply:
There's an edge-case that we may hit if we set the finalized and justified epochs the same.
(I ran into it syncing medalla)
There will be a period upon restore where we will be feeding in blocks, but those blocks will not be the 'head' of the chain. This will be the case until we process the block that actually finalized the epoch of the finalized epoch we restore with.
The issue we can run into is that while we are getting to this epoch, we may create a head that triggers ERR_INVALID_BEST_NODE, which can happen if the best node (which falls back to the justified node) is not viable for head (doesn't have the same finalized and justified and finalized epochs as is stored in the fork choice). This happens when we set the justified checkpoint, but don't set the finalized checkpoint, and then the justified node no longer matches the fork choice justified and finalized epochs.

Reply:
can you add this as comment explaining +1 :)